components/post_edit_history/restore_post_modal.tsx

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useRef} from 'react';
+import {Modal} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+import {useSelector} from 'react-redux';
+
+import {Post} from '@mattermost/types/posts';
+import {isModalOpen} from 'selectors/views/modals';
+import {GlobalState} from 'types/store';
+import {ModalIdentifiers} from 'utils/constants';
+
+type Props = {
+    post: Post;
+    actions: {
+        handleRestore: (post: Post) => void;
+        closeModal: (modalId: string) => void;
+    };
+}
+
+const RestorePostModal = ({post, actions}: Props) => {
+    const restorePostBtn = useRef<HTMLButtonElement>(null);
+    const show = useSelector((state: GlobalState) => isModalOpen(state, ModalIdentifiers.RESTORE_POST_MODAL));
+
+    const onHide = () => actions.closeModal(ModalIdentifiers.RESTORE_POST_MODAL);
+    const handleEntered = () => restorePostBtn?.current?.focus();
+
+    const handleRestore = async () => {
+        await actions.handleRestore(post);
+        onHide();
+    };
+
+    return (
+        <Modal
+            dialogClassName='a11y__modal channel-invite'
+            show={show}
+            onEntered={handleEntered}
+            onHide={onHide}
+            enforceFocus={false}
+            id='restorePostModal'
+            role='dialog'
+            aria-labelledby='restorePostModalLabel'
+        >
+            <Modal.Header closeButton={true}/>
+            <Modal.Body
+                role='application'
+                className='overflow--visible'
+            >
+                <div className='edit-post__restore__modal__header'>
+                    <FormattedMessage
+                        id='restore_post.title'
+                        defaultMessage='Restore this message?'
+                    />
```

Comment:
0/5, but I like moving these to a const outside of the component, defined with `const modalMessages = defineMessages({` and replacing these with 
`{formatMessage(modalMessages.title)}`

Reply:
Updated as suggested